include/tvm/runtime/object.h

```diff
@@ -372,8 +372,8 @@ class ObjectPtr {
   template <typename U>
   ObjectPtr(const ObjectPtr<U>& other)  // NOLINT(*)
       : ObjectPtr(other.data_) {
-    static_assert(std::is_base_of<T, U>::value,
-                  "can only assign of child class ObjectPtr to parent");
```

Comment:
Likely we want to keep this static assert

Reply:
Agree with TQ this should not be remove